### ISSUE TITLE
Clean up the customization system

### ISSUE THREAD

**Issue by unlocomqx** [PrestaShop/PrestaShop#12264]
Hi,

As the customization system was changed in PrestaShop 1.7 (#5341), a lot of "backward compatibility" code is still out there and it's causing some price display bugs (screenshot)
![2019-01-22_12-36](https://user-images.githubusercontent.com/793712/51533159-57218480-1e42-11e9-9994-1c1ecc1a94c4.png)
For example, this is caused by the line https://github.com/PrestaShop/PrestaShop/blob/7ebb37fa155f75a5600f482d8a15ab5bda3f0a9e/classes/Product.php#L5167

The reason for this display bug is that `total_customization_wt` should be used for a customized product not `total_wt`

And for this case, the test (if the product is customized) should be made in this line
https://github.com/PrestaShop/PrestaShop/blob/7ebb37fa155f75a5600f482d8a15ab5bda3f0a9e/src/Adapter/Presenter/Order/OrderLazyArray.php#L178

But with the new system, we shouldn't really care because the product row could already include the customization cost and we don't have to test if it's customized or not, thus making the code simpler

IMO, the backward compatibility code is no longer necessary and cleaning it up will make the code much cleaner and much more maintainable

We won't have to worry about customizations quantities or prices as each customization will have it's own product row in the cart, order, invoice etc

Basically, we can remove/deprecate the method Product::addCustomizationPrice it would be no longer necessary 
Also, the templates will be a lot simpler with the new logic

If you'd agree, I can make a PR to remove some of the old code and we can further discuss the changes 

Cheers

**Comment by khouloudbelguith** [comment]
Hi @unlocomqx,

We manage to reproduce the issue, it is added to our debug roadmap.
Your ticket is a duplicate of #10912
So, I close it.
Thanks!
about your fix, would you be willing to make a pull request on GitHub with your code suggestion?
https://github.com/PrestaShop/PrestaShop/tree/develop
Our developers will help you getting it right
Thank you!

**Comment by khouloudbelguith** [comment]
Duplicate of #10912

**Comment by unlocomqx** [comment]
@khouloudbelguith Thank you, I will submit a PR soon
